src/main/kotlin/org/gradle/script/lang/kotlin/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -59,9 +59,9 @@ class KotlinBuildScriptCompiler(
     val gradleApiExtensions: ClassPath,
     gradleScriptKotlinJars: ClassPath) {
 
-    val scriptResource = scriptSource.resource!!
-    val scriptFile = scriptResource.file!!
-    val script = convertLineSeparators(scriptResource.text!!)
+    val scriptFilePath = scriptSource.fileName!!
+    val scriptFile = File(scriptFilePath)
```

Comment:
From what I understood, `scriptSource.fileName` doesn't necessarily point to an existing file, right?

If that's the case we shouldn't be creating a `File` out of it as it might be used in cache keys later on (see `compilePluginsBlockOf`).